components/apps_modal/apps_modal.tsx

```diff
@@ -0,0 +1,242 @@
+
+import React from 'react';
+
+import {AppCall, AppCallResponse, AppField, AppForm, AppModalState} from 'mattermost-redux/types/apps';
+import {InteractiveDialogConfig, DialogElement} from 'mattermost-redux/types/integrations';
+
+import EmojiMap from 'utils/emoji_map';
+
+import InteractiveDialog from './interactive_dialog';
+
+type FormValues = {
+    [name: string]: any;
+};
+
+type Props = {
+    modal?: AppModalState;
+    onHide: () => void;
+    actions: {
+        doAppCall: (call: AppCall) => Promise<{data: AppCallResponse}>;
+    };
+    emojiMap: EmojiMap;
+};
+
+const AppsModal: React.FC<Props> = (props: Props) => {
+    if (!props.modal) {
+        return null;
+    }
+
+    const submitDialog = async (submission: {values: FormValues}): Promise<{data: AppCallResponse}> => {
+        if (!props.modal) {
+            return {data: {type: 'error', error: 'props.modal is not defined'}};
+        }
+
+        const call: AppCall = {
+            ...props.modal.call,
+            type: '',
+            values: submission.values,
+        };
+
+        try {
+            const res = await props.actions.doAppCall(call);
+            return res;
+        } catch (e) {
+            return {data: {type: 'error', error: e.message}};
+        }
+    };
+
+    const onHide = () => {
+        return props.onHide();
+    };
+
+    const {form, call} = props.modal;
+    const f = {
+        ...form,
+        app_id: call.context.app_id,
+        url: call.url || '',
+    };
+    const dialog = appFormToInteractiveDialog(f);
+    const dialogProps = makeDialogProps(dialog);
+
+    return (
+        <InteractiveDialog
+            {...dialogProps}
+            modal={props.modal}
+            onHide={onHide}
+            emojiMap={props.emojiMap}
+            actions={{
+                submit: submitDialog,
+            }}
+        />
+    );
+};
+
+export default AppsModal;
+
+function makeDialogProps(data: InteractiveDialogConfig) {
+    return {
+        url: data.url,
+        callbackId: data.dialog.callback_id,
+        elements: data.dialog.elements,
+        title: data.dialog.title,
+        introductionText: data.dialog.introduction_text,
+        iconUrl: data.dialog.icon_url,
+        submitLabel: data.dialog.submit_label,
+        notifyOnCancel: data.dialog.notify_on_cancel,
+        state: data.dialog.state,
+        appID: data.app_id,
+    };
+}
+
+export function appFormToInteractiveDialog(form: AppForm & {app_id: string; url: string}): InteractiveDialogConfig {
+    const fields = form.fields || [];
+    const elements = fields.map(fieldToDialogElement);
+    const config: InteractiveDialogConfig = {
+        app_id: form.app_id,
+        url: form.url,
+        trigger_id: '',
+        dialog: {
+            title: form.title || '',
+            elements,
+            introduction_text: form.header || '',
+            state: '',
+            submit_label: '',
+            notify_on_cancel: false,
```

Comment:
I wonder if we want to add this functionality eventually.

Reply:
Made a ticket https://mattermost.atlassian.net/browse/MM-30588